use nomad to deploy

kayiwa opened the pull request:
- **we plan to use Docker to render this**
- **add the deployment script**

related to https://github.com/pulibrary/princeton_ansible/issues/5787

Instead of using static sites. Use the nomad/consul platform

kayiwa committed:
we plan to use Docker to render this

Add an nginx Dockerfile
Add a Docker-compose yaml file
Improve the README to reflect these changes

Co-authored-by: Beck Davis <beck-davis@users.noreply.github.com>
Co-authored-by: Anna Headley <hackartisan@users.noreply.github.com>

kayiwa committed:
add the deployment script

kayiwa commented:
The deploy expects main. Hoping you can provide some insight on how to make this work @tpendragon

tpendragon commented:
@kayiwa I think you just need to build the docker image on commit. Add something like this: https://github.com/pulibrary/dpul-collections/blob/main/.github/workflows/build-docker-image.yml

You can get rid of the container scanning (https://github.com/pulibrary/dpul-collections/blob/main/.github/workflows/build-docker-image.yml#L76-L149) if you're not worried about that.

Once it's built you can deploy with `BRANCH=i5787_static_site bin/deploy staging`

Then you'll need something on the load balancer like https://github.com/pulibrary/princeton_ansible/blob/main/roles/nginxplus/files/conf/http/dev/abid_staging.conf with https://github.com/pulibrary/princeton_ansible/blob/main/roles/nginxplus/files/conf/http/dev/abid_staging.conf#L6 changed.

kayiwa commented on staging.hcl:
```diff
@@ -0,0 +1,31 @@
+  group "web" {
+    count = 2
+    network {
+      port "http" { to = 3000 }
```
say port 80

tpendragon commented on staging.hcl:
```diff
@@ -0,0 +1,31 @@
+        ports = ["http"]
+        force_pull = true
+      }
+      resources {
```
I'd just get rid of this resources block. This says you need 1 ghz and 500 mb of memory for this static site, and I just kinda doubt that. That's what we have for our rails apps.

kayiwa committed:
add vulnerability scanning and auto docker build

we add a vulnerability scan that runs nightly. If it catches anything it
will create an issue
we will also build our images in-situ
Co-authored-by: Angel Ruiz <aruiz1789@users.noreply.github.com>
Co-authored-by: Beck Davis <beck-davis@users.noreply.github.com>
Co-authored-by: Trey Pendragon <tpendragon@users.noreply.github.com>

kayiwa committed:
use port 80

Co-authored-by: Angel Ruiz <aruiz1789@users.noreply.github.com>
Co-authored-by: Beck Davis <beck-davis@users.noreply.github.com>
Co-authored-by: Trey Pendragon <tpendragon@users.noreply.github.com>

kayiwa committed:
we do not recommend force pushing
